src/FunctionApps/python/IntakePipeline/conversion.py

```diff
@@ -1,18 +1,62 @@
 import logging
 import re
 import requests
+import hl7
 from typing import List, Dict
 
 
-def clean_message(message: str, delimiter: str = "\n") -> str:
-    cleaned_message = re.sub("[\r\n]+", delimiter, message)
+def clean_message(message: str) -> str:
+    parsed_message = None
+    try:
+        parsed_message = hl7.parse(message)
+
+        # MSH-7
+        cleansed_datetime = normalize_hl7_datetime(parsed_message[0][7][0])
+        parsed_message[0][7][0] = cleansed_datetime
```

Comment:
How confident are we that the datetime that was causing the issue will always occur in that position? I'm a little nervous of there being other dates buried in the messages, or even of the structure of the message being such that some other information content is in this place instead of a datetime.